cloud and parallel track selection

aehyvari opened the pull request.

jhoenicke commented:
`tools/prep/aws_make_jobs.py` is missing in the pull request

jhoenicke commented on 2021/prep/selection/aws_make_selection.sh:
```diff
@@ -0,0 +1,139 @@
+
+# Note that python2 and python3 disagree on random choice function.
+# Always use python3 to get reproducible results.
+SEED=332349782
```
We may want to put the SEED somewhere at the beginning or in an extra file.  See also the `make-selection.sh` script

aehyvari replied:
I was thinking of the same.  I'll commit some suggestion here then.

aehyvari replied:
I made a separate file `../../COMPETITION_SEED` (deliberately not part of this commit) and changed both `aws_make_selection.sh` and `make_selection.sh` to use the that file with some checks about the correctness of the seed.

jhoenicke reviewed (commented):
Can you add a short README with how to run the scripts, the expected directory layout, and what is generated?

aehyvari commented on 2021/prep/selection/aws_scramble_and_rename.sh:
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+SCRAMBLER=~/bin/scrambler
```
That makes sense and is pretty natural.

aehyvari commented on 2021/prep/selection/aws_make_selection.sh:
```diff
@@ -0,0 +1,139 @@
+PYTHON=python3
+
+if [ $# != 2 ]; then
+    echo "Usage: $0 <smt-lib-root> <competition-root>"
```
That could be good.  Do you think the selection could grow so big that it could become a problem?  The output directory contains the readily scrambled instances, not just the names of them like in the normal selection.

jhoenicke replied:
I didn't think about this. Probably worth mentioning in the readme.  I think even with hardcoded pathts one could use symlinks to force it to another disk.

aehyvari replied:
I'll add it to the readme but keep this as it is unless you object.

aehyvari commented on 2021/prep/selection/aws_make_selection.sh:
```diff
@@ -0,0 +1,139 @@
+
+selectfinal ${SELECTION_NUMBERS_PARALLEL} ${OUT_PARALLEL}-hard ${OUT_PARALLEL}-unsolved ${SELECTION_PARALLEL}
+
+${AWS_SCRAMBLER} ${smt_lib_root} \
```
Ok.

aehyvari commented on tools/selection/selection_additive.py:
```diff
@@ -0,0 +1,244 @@
+                    help='The prefix to prepend to selected benchmark lines')
+    ap.add_argument('--selection-mode', dest='selection_mode',
+                    default='hard-only', help='Which type of benhmarks are included'\
+                    'Either hard-only or unsoved-only')
```
Fixed.

aehyvari commented on 2021/prep/selection/aws_make_selection.sh:
```diff
@@ -0,0 +1,139 @@
+CLOUD_LOGICS="UF;UFDT;ALIA;AUFLIA;UFLIA;UFIDL;AUFLIRA;UFLRA;UFDTLIA;UFDTLIRA;AUFDTLIA;AUFDTLIRA;AUFDTNIRA;UFDTNIA;UFDTNIRA;AUFNIA;AUFNIRA;UFNIA;LRA;LIA;NIA;NRA;BV;QF_ABV;QF_BV;QF_BVFP;QF_FP;QF_IDL;QF_LIA;QF_LIRA;QF_LRA;QF_NIA;QF_NRA;QF_UF;QF_UFNRA;UFBV;QF_RDL"
+PARALLEL_LOGICS="UF;UFDT;ALIA;AUFLIA;UFLIA;UFIDL;AUFLIRA;UFLRA;UFDTLIA;UFDTLIRA;AUFDTLIA;AUFDTLIRA;AUFDTNIRA;UFDTNIA;UFDTNIRA;AUFNIA;AUFNIRA;UFNIA;LRA;LIA;NIA;NRA;BV;QF_ABV;QF_BV;QF_BVFP;QF_FP;QF_IDL;QF_LIA;QF_LIRA;QF_LRA;QF_NIA;QF_NRA;QF_UF;QF_UFNRA;UFBV"
+
+RATIO=1.0
```
No, they should be removed.

aehyvari commented:
I think that I addressed the comments.  The only one I didn't do was to hard-code the output benchmarks.  A typical selection size seems to be something like a little less than a gigabyte.  If you think that's ok, I can just create the instances to the working directory.

jhoenicke commented:
Just an observation:  currently the code uses roughly the same benchmarks for cloud and parallel track (modulo some changes because of different competitive logics).  One could change this, but it may be advantageous if the tracks are comparable by running the same benchmarks.

aehyvari commented:
I'm going to do some cleaning so that the code doesn't leave temporary files to the filesystem.
